<?xml version="1.0"?>
<launch>
  <arg name="chain_start" default="base_link" />
  <arg name="chain_end" default="tcp" />
  <arg name="timeout" default="0.05" />
  <arg name="sim" default="true"/>

  <include file="$(find mr_description)/launch/manipulator5d_display.launch" >
    <arg name="sim" value="false"/>
  </include>
  
  <group if="$(arg sim)">
    <node name="interactive_marker" pkg="modular_robot_control" type="eef_interactive_marker.py" args="'base_link' 'tcp'"/>
    <node name="set_rosparam_joint_names" pkg="modular_robot_control" type="rosparam_set_joint_names.py"/>
    <node name="ik_node" pkg="modular_robot_control" type="ik_node" output="screen" args="true">
      <param name="chain_start" value="$(arg chain_start)"/>
      <param name="chain_end" value="$(arg chain_end)"/>
      <param name="timeout" value="$(arg timeout)"/>
      <param name="urdf_param" value="/robot_description"/>
      <remap from="/pose" to="/marker_pose"/>
    </node>
  </group>

  <group unless="$(arg sim)">
    <node name="interactive_marker" pkg="modular_robot_control" type="eef_interactive_marker.py" args="'base_link' 'tcp'"/>
    <node name="set_rosparam_joint_names" pkg="modular_robot_control" type="rosparam_set_joint_names.py"/>
    <node name="ik_node" pkg="modular_robot_control" type="ik_node" output="screen" args="false">
      <param name="chain_start" value="$(arg chain_start)"/>
      <param name="chain_end" value="$(arg chain_end)"/>
      <param name="timeout" value="$(arg timeout)"/>
      <param name="urdf_param" value="/robot_description"/>
      <remap from="/pose" to="/marker_pose"/>
    </node>
    <node name="canopen_driver" pkg="canopen_motor_node" type="canopen_motor_node" output="screen" clear_params="true">
					<!--   launch-prefix="gdb -ex run -/-args">-->
      <rosparam command="load" file="$(find birl_module_canopen)/config/can.yaml" />
      <rosparam command="load" file="$(find birl_module_canopen)/config/manipulator5d_motors.yaml" />
    </node>
<!-- controllers -->
    <rosparam command="load" file="$(find modular_robot_control)/config/manipulator5d_controller.yaml" />
   <!-- start_controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="controller_manager" args="spawn joint_state_controller joint_group_position_controller" respawn="false" output="screen"/>
<!--  <node name="controller_loader" pkg="controller_manager" type="controller_manager" args="load joint_trajectory_controller I_Joint_position_controller I_Joint_velocity_controller " respawn="false" output="screen"/>-->  
<!--  transform the topic from joint_state_publisher to that of canopen_motor_node-->
<!--  <node name="topic_transf_joint_state_pub_to_canopen_motor" pkg="topic_tools" type="transform" args="/joint_states /I_Joint_position_controller/command std_msgs/Float64 'm.position[0]'" output="screen" />-->
    <node name="module_state_adaptor" pkg="birl_module_canopen" type="module_state_adaptor.py" args="canopen_driver">
      <remap from="joint_states" to="/joint_command"/>
    </node>
  </group>

</launch>